src/qibo/models.py

```diff
@@ -1,10 +1,12 @@
 import math
 from qibo import get_backend
-from qibo.config import raise_error
+from qibo.config import raise_error, log
 from qibo.core.circuit import Circuit as StateCircuit
 from qibo.core.circuit import DensityMatrixCircuit
 from qibo.evolution import StateEvolution, AdiabaticEvolution
 from typing import Dict, Optional
+import numpy as np
+from qibo import gates
```

Comment:
I would suggest to perform this imports inside the Grover class, to avoid importing overhead for other models.

Reply:
@igres26, following our discussion, if you move these imports inside the class, you have to use `self.np` and `self.gates` in order to access them in the class methods. I had a quick look at the code and there aren't many calls to gates and np so this seems an acceptable solution to me. 

Something additional to keep in mind. If you do:
```Python
class Grover:
    import numpy as np
    from qibo import gates
    ...
```
I don't think there is a big difference with doing global imports in terms of overhead because I believe that Python will still import these libraries if the user does `from qibo import models`. On the other hand if you do:
```Python
class Grover:
    
    def __init__(self, ...):
        import numpy as np
        self.np = np
```
then numpy will be imported only when a Grover model is initialized. In both cases you have to do `self.np` to use numpy later in the model. This note is true for existing models too, not only Grover.

Reply:
For now I have followed @stavros11 advice on importing inside the class.